src/libstd/io/stdio.rs

```diff
@@ -701,6 +705,14 @@ pub fn _print(args: fmt::Arguments) {
     print_to(args, &LOCAL_STDOUT, stdout, "stdout");
 }
 
+#[unstable(feature = "print_internals",
+           reason = "implementation detail which may disappear or be replaced at any time",
+           issue = "0")]
+#[doc(hidden)]
+pub fn _try_print(args: fmt::Arguments) -> io::Result<()> {
+    try_print_to(args, &LOCAL_STDOUT, stdout)
+}
```

Comment:
seems fine, but I feel like we could easily use stable APIs to build this, no? Like, can't we just make `librustc_driver` use `write!` or something?

Reply:
The primary reason I did it this way was to make sure `try_println!` had ensured identical behaviour as `println!` (apart from the panic behaviour). I'm not sure how important this is in practice, but felt that it was generally a good idea to keep things familiar, so as not to introduce some inconsistency that would lead to a different issue in the future. (Plus, if we ever want to make such functionality public, it's trivial.)

If you think this is overcomplicating things, though, I could `write` inside `librustc_driver` instead, yeah.